### PR TITLE
created exclude-time-filter class

### PR DESCRIPTION
**Pull request by sugolov**
ExcludeTimeFilter for excluding time slots. Implementation almost identical to TimeFilter, maybe there is a way to merge them? If there is, I can handle doing so. Will create test cases soon.

### CODE REVIEW

**Comment by sugolov** [comment]
filter is pretty broken rn. tests might be as well

i will fix this soon, but main changes are moving the day enum

File: src/main/java/filters/ExcludeTimeFilter.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     * @return Whether some section conflicts with the time range
+     */
+    @Override
+    public boolean checkSchedule(Schedule s) {
+        if (s == null) {
+            return false;
+        }
+        List<Timeslot> timeslots = new ArrayList();
+
+        for (Section lec : s.getLectures()) {
```
Comment (sugolov) [inline src/main/java/filters/ExcludeTimeFilter.java:54]: tbh this is repeated code and occurs in multiple filters. siddarth's pr adds a method to get all these slots, so when he merges it we should add that in everywehre

**Comment by sugolov** [comment]
should be working now. let me know if i have missed anything

File: src/main/java/filters/ExcludeTimeFilter.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * This class filter checks whether a course overlaps with a specific time. The schedule does not pass the filter if
+ * it overlaps.
+ */
+public class ExcludeTimeFilter implements Filter {
+
+    private LocalTime lowerBound;
+    private LocalTime upperBound;
+    private Day filteredDay;
+
+    /**
```
Comment (kenneth-miura) [inline src/main/java/filters/ExcludeTimeFilter.java:22]: beast javadoc

Comment (kenneth-miura) [reply src/main/java/filters/ExcludeTimeFilter.java:22]: ＼(＾O＾)／

**Review by kenneth-miura — APPROVED** [review]
looks good to me

File: src/main/java/filters/ExcludeTimeFilter.java (hunk 2 above)
Comment (sugolov) [reply src/main/java/filters/ExcludeTimeFilter.java:22]: thanks kang

**Pull request merged**